conn_go19_test.go

```diff
@@ -0,0 +1,67 @@
+//go:build go1.9
+// +build go1.9
+
+package pq
+
+import (
+	"reflect"
+	"testing"
+)
+
+func TestArrayArg(t *testing.T) {
+	db := openTestConn(t)
+	defer db.Close()
+
+	for _, tc := range []struct {
+		name string
+		want interface{}
+	}{
+		{
+			name: "slice-value",
+			want: []int64{245, 231},
+		},
+		{
+			name: "slice-pointer",
+			want: &[]int64{245, 231},
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			r, err := db.Query("SELECT $1::int[]", tc.want)
+			if err != nil {
+				t.Fatal(err)
+			}
+			defer r.Close()
+
+			if !r.Next() {
+				if r.Err() != nil {
+					t.Fatal(r.Err())
+				}
+				t.Fatal("expected row")
+			}
+
+			defer func() {
+				if r.Next() {
+					t.Fatal("unexpected row")
+				}
+			}()
+
+			rt := reflect.TypeOf(tc.want)
+			wantPtr := rt.Kind() == reflect.Ptr
+			if wantPtr {
+				rt = rt.Elem()
+			}
+			got := reflect.New(rt)
+			if err := r.Scan(Array(got.Interface())); err != nil {
```

Comment:
I recall more types being tested before. If the input and output types can't match, perhaps add another field in the test case, `in, out interface{}`.

```
in:  []int{245, 231},
out: []int64{245, 231},
```